diff --git a/dist/index.mjs b/dist/index.mjs
index f664fc1f4af47dc5a445d1c42f45fe12d5dbb191..fde3b5b5ebd150a181c1e0243b5b747c6b022c8a 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -1,6 +1,7 @@
 import { Language, Parser } from "web-tree-sitter";
 import { util } from "prettier";
 import { builders, utils } from "prettier/doc";
+import treeSitterJavaOrchard from './tree-sitter-java_orchard.wasm';
 //#region src/options.ts
 var options_default = {
 	arrowParens: {
@@ -2381,7 +2382,7 @@ var parser_default = {
 const parser = (async () => {
 	await Parser.init();
 	const parser = new Parser();
-	const Java = await Language.load(new URL("./tree-sitter-java_orchard.wasm", import.meta.url));
+	const Java = await Language.load(treeSitterJavaOrchard);
 	parser.setLanguage(Java);
 	return parser;
 })();
