diff --git a/dist/index.mjs b/dist/index.mjs
index ad2498336527397914f097450cb26503c13de9b7..a48ab42508d648d2618d1fb1f08124edc6adac5a 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -1,6 +1,7 @@
 import { Language, Parser } from "web-tree-sitter";
 import { util } from "prettier";
 import { builders, utils } from "prettier/doc";
+import treeSitterJavaOrchard from './tree-sitter-java_orchard.wasm';
 //#region src/options.ts
 var options_default = {
 	arrowParens: {
@@ -2360,7 +2361,7 @@ var parser_default = {
 const parser = (async () => {
 	await Parser.init();
 	const parser = new Parser();
-	const Java = await Language.load(new URL("./tree-sitter-java_orchard.wasm", import.meta.url));
+	const Java = await Language.load(treeSitterJavaOrchard);
 	parser.setLanguage(Java);
 	return parser;
 })();
