diff --git a/web-tree-sitter.js b/web-tree-sitter.js
index 449545cbe28eabba64252ea5097a7d8f0a686ba8..7ee945d4ebc60b5b57b33ba05a191e204a37ec32 100644
--- a/web-tree-sitter.js
+++ b/web-tree-sitter.js
@@ -1,3 +1,5 @@
+import treeSitterWasmBytes from './web-tree-sitter.wasm';
+
 var __defProp = Object.defineProperty;
 var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
 
@@ -1484,8 +1486,7 @@ var Language = class _Language {
     if (input instanceof Uint8Array) {
       binary2 = input;
     } else if (globalThis.process?.versions.node) {
-      const fs2 = await import("fs/promises");
-      binary2 = await fs2.readFile(input);
+      throw new Error('Not implemented.');
     } else {
       const response = await fetch(input);
       if (!response.ok) {
@@ -1522,8 +1523,7 @@ async function Module2(moduleArg = {}) {
   var Module = moduleArg;
   var ENVIRONMENT_IS_WEB = typeof window == "object";
   var ENVIRONMENT_IS_WORKER = typeof WorkerGlobalScope != "undefined";
-  var ENVIRONMENT_IS_NODE = typeof process == "object" && process.versions?.node && process.type != "renderer";
-  if (ENVIRONMENT_IS_NODE) {
+  if (false) {
     const { createRequire } = await import("module");
     var require = createRequire(import.meta.url);
   }
@@ -1546,7 +1546,7 @@ async function Module2(moduleArg = {}) {
   }
   __name(locateFile, "locateFile");
   var readAsync, readBinary;
-  if (ENVIRONMENT_IS_NODE) {
+  if (false) {
     var fs = require("fs");
     if (_scriptName.startsWith("file:")) {
       scriptDirectory = require("path").dirname(require("url").fileURLToPath(_scriptName)) + "/";
@@ -1621,7 +1621,7 @@ async function Module2(moduleArg = {}) {
   var wasmBinary;
   var ABORT = false;
   var EXITSTATUS;
-  var isFileURI = /* @__PURE__ */ __name((filename) => filename.startsWith("file://"), "isFileURI");
+  var isFileURI = /* @__PURE__ */ __name((filename) => typeof filename === 'string' && filename.startsWith("file://"), "isFileURI");
   var readyPromiseResolve, readyPromiseReject;
   var wasmMemory;
   var HEAP8, HEAPU8, HEAP16, HEAPU16, HEAP32, HEAPU32, HEAPF32, HEAPF64;
@@ -1706,6 +1706,7 @@ async function Module2(moduleArg = {}) {
   __name(abort, "abort");
   var wasmBinaryFile;
   function findWasmBinary() {
+    return treeSitterWasmBytes;
     if (Module["locateFile"]) {
       return locateFile("web-tree-sitter.wasm");
     }
@@ -1735,8 +1736,7 @@ async function Module2(moduleArg = {}) {
   __name(getWasmBinary, "getWasmBinary");
   async function instantiateArrayBuffer(binaryFile, imports) {
     try {
-      var binary2 = await getWasmBinary(binaryFile);
-      var instance2 = await WebAssembly.instantiate(binary2, imports);
+      var instance2 = await WebAssembly.instantiate(binaryFile, imports);
       return instance2;
     } catch (reason) {
       err(`failed to asynchronously prepare wasm: ${reason}`);
@@ -1745,7 +1745,7 @@ async function Module2(moduleArg = {}) {
   }
   __name(instantiateArrayBuffer, "instantiateArrayBuffer");
   async function instantiateAsync(binary2, binaryFile, imports) {
-    if (!binary2 && !isFileURI(binaryFile) && !ENVIRONMENT_IS_NODE) {
+    if (!binary2 && !isFileURI(binaryFile) && !false) {
       try {
         var response = fetch(binaryFile, {
           credentials: "same-origin"
@@ -1798,7 +1798,7 @@ async function Module2(moduleArg = {}) {
       });
     }
     wasmBinaryFile ??= findWasmBinary();
-    var result = await instantiateAsync(wasmBinary, wasmBinaryFile, info2);
+    var result = await instantiateArrayBuffer(wasmBinaryFile, info2);
     var exports = receiveInstantiationResult(result);
     return exports;
   }
