diff --git a/web-tree-sitter.js b/web-tree-sitter.js
index 449545cbe28eabba64252ea5097a7d8f0a686ba8..4f64811560537a23997e14749ae09f38a25d0d1f 100644
--- a/web-tree-sitter.js
+++ b/web-tree-sitter.js
@@ -1,6 +1,22 @@
 var __defProp = Object.defineProperty;
 var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
 
+function decodeBase64DataUri(url) {
+  if (url instanceof URL) url = url.href;
+  else if (typeof url !== 'string') return undefined;
+
+  const b64index = url.indexOf('base64,');
+  if (b64index !== -1 && url.startsWith('data:')) {
+    const decoded = atob(url.slice(b64index + 7));
+    const bytes = new Uint8Array(decoded.length);
+    for (let i = 0; i < decoded.length; i++) {
+        bytes[i] = decoded.charCodeAt(i);
+    }
+    return bytes;
+  }
+  return undefined;
+}
+
 // src/edit.ts
 var Edit = class {
   static {
@@ -1481,11 +1497,14 @@ var Language = class _Language {
    */
   static async load(input) {
     let binary2;
-    if (input instanceof Uint8Array) {
+    const bytes = decodeBase64DataUri(input);
+
+    if (bytes !== undefined) {
+      binary2 = bytes;
+    } else if (input instanceof Uint8Array) {
       binary2 = input;
     } else if (globalThis.process?.versions.node) {
-      const fs2 = await import("fs/promises");
-      binary2 = await fs2.readFile(input);
+      throw new Error('Not implemented.');
     } else {
       const response = await fetch(input);
       if (!response.ok) {
@@ -1522,8 +1541,7 @@ async function Module2(moduleArg = {}) {
   var Module = moduleArg;
   var ENVIRONMENT_IS_WEB = typeof window == "object";
   var ENVIRONMENT_IS_WORKER = typeof WorkerGlobalScope != "undefined";
-  var ENVIRONMENT_IS_NODE = typeof process == "object" && process.versions?.node && process.type != "renderer";
-  if (ENVIRONMENT_IS_NODE) {
+  if (false) {
     const { createRequire } = await import("module");
     var require = createRequire(import.meta.url);
   }
@@ -1546,7 +1564,7 @@ async function Module2(moduleArg = {}) {
   }
   __name(locateFile, "locateFile");
   var readAsync, readBinary;
-  if (ENVIRONMENT_IS_NODE) {
+  if (false) {
     var fs = require("fs");
     if (_scriptName.startsWith("file:")) {
       scriptDirectory = require("path").dirname(require("url").fileURLToPath(_scriptName)) + "/";
@@ -1745,7 +1763,12 @@ async function Module2(moduleArg = {}) {
   }
   __name(instantiateArrayBuffer, "instantiateArrayBuffer");
   async function instantiateAsync(binary2, binaryFile, imports) {
-    if (!binary2 && !isFileURI(binaryFile) && !ENVIRONMENT_IS_NODE) {
+    if (!binary2 && !isFileURI(binaryFile) && !false) {
+      const bytes = decodeBase64DataUri(binaryFile);
+      if (bytes !== undefined) {
+        return await WebAssembly.instantiate(bytes, imports);
+      }
+
       try {
         var response = fetch(binaryFile, {
           credentials: "same-origin"
